provider/sql.go

```diff
@@ -429,17 +437,17 @@ func (mat *sqlMaterialization) IterateSegment(start, end int64) (FeatureIterator
 	query := mat.query.materializationIterateSegment(mat.tableName)
 	rows, err := mat.db.Query(query, start, end)
 	if err != nil {
-		return nil, err
+		return nil, fferr.NewExecutionError(string(mat.providerType), "", "", "", err)
 	}
 	types, err := rows.ColumnTypes()
 	if err != nil {
-		return nil, err
+		return nil, fferr.NewExecutionError(string(mat.providerType), "", "", "", err)
```

Comment:
this should probably be thrown inside ColumnTypes()

Reply:
This `ColumnTypes` isn't ours, hence why I wrapped it.